light-curve/light_curve/light_curve_py/features/rainbow.py

```diff
@@ -0,0 +1,214 @@
+        t0, T_min, delta_T, k_sig = params[self.temp_params_idx]
+        dt = t - t0
+        return T_min + delta_T / (1.0 + np.exp(dt / k_sig))
+
+    @staticmethod
+    def planck_nu(wave_cm, T):
+        """Planck function in frequency units."""
+        nu = speed_of_light / wave_cm
+        return (
+            (2 * planck_constant / speed_of_light**2)
+            * nu**3
+            / np.expm1(planck_constant * nu / (boltzman_constant * T))
+        )
+
+    def normalized_intensity(self, wave_cm, T):
+        return self.planck_nu(wave_cm, T) / self.planck_nu_norm
+
+    def _lsq_model(self, x, *params):
+        """Model function for the fit."""
+        t, band_idx, wave_cm = x
+        params = np.array(list(params))
+
+        lum = self.lum_func(t, params)
+        T = self.temp_func(t, params)
+        flux = lum * self.normalized_intensity(wave_cm, T)
+        if self.with_baseline:
+            baseline_param_idx = band_idx + len(P)
+            baselines = params[baseline_param_idx]
+            flux += baselines
+        return flux
+
+    def model(self, t, band, *params):
+        """Model function for the fit."""
+        band_idx = self.lookup_band_idx(band)
+        wave_cm = self.wave_cm_array[band_idx]
+        return self._lsq_model((t, band_idx, wave_cm), *params)
+
+    def _eval(self, *, t, m, sigma, band):
+        try:
+            from iminuit import Minuit
+
+            if LeastSquares is None:
+                raise ImportError("iminuit is required for RainbowFit feature, please install it")
+        except ImportError:
+            if sys.version_info < (3, 8):
+                raise ImportError(
+                    "iminuit is required for RainbowFit feature, "
+                    "and the required version is not available for Python < 3.8"
+                )
+            raise ImportError("iminuit is required for RainbowFit feature, please install it")
+
+        # normalize input data
+        t_mean = np.mean(t)
+        t_std = np.std(t)
+        t = (t - t_mean) / t_std
+        m_std = np.std(m)
+        m = m / m_std
+        sigma = sigma / m_std
+
+        t_amplitude = t[-1] - t[0]
+        m_amplitude = np.ptp(m)
+
+        initial_guesses = {
+            "reference_time": 0.0,
+            "amplitude": 1.0,
+            "rise_time": 1.0,
+            "fall_time": 1.0,
+            "Tmin": 4000.0,
+            "delta_T": 7000.0,
+            "k_sig": 1.0,
+        }
+        limits = {
+            "reference_time": (t[0] - 10 * t_amplitude, t[-1] + 10 * t_amplitude),
+            "amplitude": (0.0, 10 * m_amplitude),
+            "rise_time": (0.0, 10 * t_amplitude),
+            "fall_time": (0.0, 10 * t_amplitude),
+            "Tmin": (1e2, 1e6),  # K
+            "delta_T": (0.0, 1e6),  # K
+            "k_sig": (0.0, 10 * t_amplitude),
+        }
+        if self.with_baseline:
+            initial_guesses.update({f"baseline_{band}": 0.0 for band in self.band_wave_cm})
```

Comment:
Shouldn't you put np.min(m) instead of 0 for the initial guess of baseline ?

Reply:
Thanks for this. There actually was an error in a different part, I've forgotten to implement flux normalization for the baseline-full case. It should be fixed now